src/Core/Services/MetadataProviders/CosmosSqlMetadataProvider.cs

```diff
@@ -570,7 +578,7 @@ public string GetEntityName(string graphQLType)
             }
 
             // Fallback to looking at the singular name of the entity.
-            foreach ((string _, Entity entity) in _runtimeConfig.Entities)
+            foreach ((string _, Entity entity) in _runtimeConfigProvider.GetConfig().Entities)
```

Comment:
Do you need to fetch the `runtimeConfig` object a second time? you already fetch at the beginning of this function.